### PR TITLE
add smooth factor flag

### PR DESCRIPTION
**Pull request by taylanbil**
This will give us the option to disable rate smoothing if we so desire.

0.8 is the default value from xla_model.py

### CODE REVIEW

**Comment by taylanbil** [comment]
this covers all the files that use `RateTracker`

```bash
taylanbil@DESKTOP:~/tmp/github/xla/test on argsmooth
% grep RateTracker *                                        [190821 | 2:35PM] ✹
grep: cpp: Is a directory
test_train_cifar.py:    tracker = xm.RateTracker(smooth_factor=FLAGS.smooth_factor)
test_train_imagenet.py:    tracker = xm.RateTracker(smooth_factor=FLAGS.smooth_factor)
test_train_mnist.py:    tracker = xm.RateTracker(smooth_factor=FLAGS.smooth_factor)
```

**Comment by dlibenzi** [comment]
Can't we choose a sane value instead?
Exposing tunables to models is OK but IMHO should be done only for meaningful cases, otherwise if we expose everything the model code becomes a bit too complex.

**Comment by taylanbil** [comment]
I was thinking one may want to not smooth at all. AFAIK that's what we do on the TF side.

**Comment by dlibenzi** [comment]
> I was thinking one may want to not smooth at all. AFAIK that's what we do on the TF side.

Does a 0.0 smoothing achieves stable results?
Can we choose a smaller one instead 0.8 as default?

**Comment by taylanbil** [comment]
I mean, sometimes we will want to smooth, sometimes not, depending on personal preference and also how variable the rate is w// the model and dataset in hand. I'm not unhappy with the current value actually. Happy to drop the pr.

**Comment by jysohn23** [comment]
> Does a 0.0 smoothing achieves stable results?
> Can we choose a smaller one instead 0.8 as default?

Do you think then setting `smoothing_factor` to 0.0 is reasonable? On the TF side we log the average examples/sec since last logging event AFAIK with TPUEstimator and would like to compare to that. Based on what I've seen, yes, 0.0 smoothing is stable.

**Comment by dlibenzi** [comment]
That depends on how frequently you call `rate()`.
If you call it at every step, a smoothing of 0.0 might not be good. Every 20 steps, probably yes.
IOW the frequency of calling `rate()` provides another smoothing factor.
I'd be inclined to choose something between, say, 0.2 and 0.6 and call it a day.

**Comment by dlibenzi** [comment]
Another approach is to leave models as they are, and instead of the usual 0.8 default, use something like:

```Python
xu.getenv_as('RATE_TRACKER_SMOOTHING', float, 0.8)
```

File: torch_xla_py/xla_model.py [hunk 1]
```diff
@@ -140,8 +140,10 @@ def xla_replication_devices(local_devices):
 
 class RateTracker(object):
 
-  def __init__(self, smooth_factor=0.8):
-    self._smooth_factor = smooth_factor
+  def __init__(self, smooth_factor=None):
+    self._smooth_factor = xu.getenv_as(
+        'RATE_TRACKER_SMOOTHING', float,
+        0.8) if smooth_factor is None else smooth_factor
```
Comment (dlibenzi) [inline torch_xla_py/xla_model.py]: Isn't this shorter?
```Python
def __init__(self, smooth_factor=xu.getenv_as('RATE_TRACKER_SMOOTHING', float, 0.8)):
```

Comment (taylanbil) [reply torch_xla_py/xla_model.py]: I thought that goes against some pep8 convention (don't call function in default value assignment)

Comment (dlibenzi) [reply torch_xla_py/xla_model.py]: SG then!

**Pull request merged**